test/test_shared.cpp

```diff
@@ -2079,14 +2108,104 @@ TEST_IF(Shared_AsyncMultiprocess, allow_async)
 
 #endif // !defined(_WIN32) && !REALM_PLATFORM_APPLE
 
-#if !defined(_WIN32)
-// this test does not work with valgrind:
-#if 0
+#ifdef _WIN32
+
+#if 1
```

Comment:
Previously, this was completely disabled on all platforms due to Valgrind (see the original comment - which I have relocated to the bottom in this PR). This PR enables it on Windows where we don't run Valgrind